homeassistant/components/mqtt/__init__.py

```diff
@@ -296,7 +296,11 @@ def async_setup(hass, config):
     if conf is None:
         conf = CONFIG_SCHEMA({DOMAIN: {}})[DOMAIN]
 
-    client_id = conf.get(CONF_CLIENT_ID)
+    # Remove the generation of the client_id after it's fixed in HBMQTT
+    import random
+    client_id = conf.get(
+        CONF_CLIENT_ID, 'home-assistant-{}'.format(random.randrange(0, 1000)))
```

Comment:
> Should we check to make sure the protocol is 3.1.1?

We could but I see no benefit. Isn't it highly unlikely that somebody depends on a random generated ID? My assumption was that if you depend on the client ID then it's configured already.

> Another option is to set the protocol to 3.1.

As far as I remember is HBMQTT only supporting MQTT 3.1.1.